src/taskpane/server.js

```diff
@@ -1128,14 +1220,27 @@ function contentFormattingSetter(sheet, model) {
 
     // protect column
     // read only fields - ie ones you can get from Spira but not create in Spira (as with IDs - eg task component)
-    if (model.fields[i].unsupported || model.fields[i].isReadOnly) {
+    if ((model.fields[i].unsupported || model.fields[i].isReadOnly) && !model.fields[i].isHidden) {
       var warning = "";
       if (model.fields[i].unsupported) {
         warning = model.fields[i].name + "unsupported";
       } else if (model.fields[i].isReadOnly) {
         warning = model.fields[i].name + " is read only";
       }
-      
+
+      protectColumn(
+        sheet,
+        columnNumber,
+        nonHeaderRows,
+        model.colors.bgReadOnly,
+        warning,
+        false
+      );
+    }
+
+    if (model.fields[i].isHidden) {

```

Comment:
add a comment above here.
Also note that both if blocks can run on a hidden field that is protected. It will protected and then hidden in two different steps. I know that may not be ideal from an Excel perspective, but will make the code cleaner and more readable.
Then the code below this calling protectColumn will be changed to hideColumn I guess

Reply:
That's true. I changed as specified.